apps/accounts/mixins.py

```diff
@@ -1,75 +1,25 @@
+from django.db.models import Model
 from django.utils import translation
-from django.utils.translation import gettext_noop
 
 from audits.const import ActionChoices
-from common.views.mixins import RecordViewLogMixin
-from common.utils import i18n_fmt
+from audits.handler import create_or_update_operate_log
 
 
-class AccountRecordViewLogMixin(RecordViewLogMixin):
+class AccountRecordViewLogMixin(object):
     get_object: callable
-    get_queryset: callable
-
-    @staticmethod
-    def _filter_params(params):
-        new_params = {}
-        need_pop_params = ('format', 'order')
-        for key, value in params.items():
-            if key in need_pop_params:
-                continue
-            if isinstance(value, list):
-                value = list(filter(None, value))
-            if value:
-                new_params[key] = value
-        return new_params
-
-    def get_resource_display(self, request):
-        query_params = dict(request.query_params)
-        params = self._filter_params(query_params)
-
-        spm_filter = params.pop("spm", None)
-
-        if not params and not spm_filter:
-            display_message = gettext_noop("Export all")
-        elif spm_filter:
-            display_message = gettext_noop("Export only selected items")
-        else:
-            query = ",".join(
-                ["%s=%s" % (key, value) for key, value in params.items()]
-            )
-            display_message = i18n_fmt(gettext_noop("Export filtered: %s"), query)
-        return display_message
-
-    @property
-    def detail_msg(self):
-        return i18n_fmt(
-            gettext_noop('User %s view/export secret'), self.request.user
-        )
-
-    def list(self, request, *args, **kwargs):
-        list_func = getattr(super(), 'list')
-        if not callable(list_func):
-            return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
-        response = list_func(request, *args, **kwargs)
-        with translation.override('en'):
-            resource_display = self.get_resource_display(request)
-            ids = [q.id for q in self.get_queryset()]
-            self.record_logs(
-                ids, ActionChoices.view, self.detail_msg, resource_display=resource_display
-            )
-        return response
+    model: Model
 
     def retrieve(self, request, *args, **kwargs):
         retrieve_func = getattr(super(), 'retrieve')
         if not callable(retrieve_func):
             return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
         response = retrieve_func(request, *args, **kwargs)
         with translation.override('en'):
-            resource = self.get_object()
-            self.record_logs(
-                [resource.id], ActionChoices.view, self.detail_msg, resource=resource
+            create_or_update_operate_log(
+                ActionChoices.view, self.model._meta.verbose_name,
+                force=True, resource=self.get_object(),
             )
         return response
 
```

Comment:
There are no significant code differences between the two versions you provided. Both use Django Rest Framework and share some of the same imports. The functionality remains mostly unchanged, but it's hard to say without testing. There is however an import error in the `AccountRecordViewLogMixin` class which should be fixed. For further optimizations and improvements, please check the full implementations rather than reviewing small diffs like this. Any changes in context or requirements beyond these base settings might warrant reworking the implementation.

If there was something specific that needed improving, fixing or differentiating from another solution, then I'd be happy to help!